Code/GraphMol/Subgraphs/Subgraphs.cpp

```diff
@@ -452,20 +455,37 @@ findAllPathsOfLengthsMtoN(const ROMol &mol, unsigned int lowerLen,
   //
   PRECONDITION(lowerLen <= upperLen, "");
 
+  double *distMat = nullptr;
```

Comment:
I think it might be more clear with a ternary

```
// the molecule owns this pointer
const double *distMat = onlyShortestPaths ? nullptr : MolOps::getDistanceMat(mol);
```

This makes it a bit more clear that we can't delete the distance matrix.

Reply:
Otherwise LGTM

Reply:
Yep, I can see that. Change made.